modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/cxf/http2/Http2EndpointTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jboss.test.ws.jaxws.cxf.http2;
+
+import jakarta.xml.ws.Service;
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.xml.namespace.QName;
+import org.apache.cxf.BusFactory;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.wsf.test.JBossWSTest;
+import org.jboss.wsf.test.JBossWSTestHelper;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class Http2EndpointTest extends JBossWSTest {
+    @ArquillianResource
+    private URL baseURL;
+
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive archive = ShrinkWrap.create(WebArchive.class, "jaxws-cxf-http2.war");
+        archive.setManifest(new StringAsset("Manifest-Version: 1.0\n"
+                        + "Dependencies: org.apache.cxf\n"))
+                .addClass(HelloWorld.class)
+                .addClass(HelloWorldImpl.class)
+                .setWebXML(new File(JBossWSTestHelper.getTestResourcesDir() + "/jaxws/cxf/http2/WEB-INF/web.xml"));
+        return archive;
+    }
+    @Test
+    @RunAsClient
+    public void testHttp11() throws Exception {
+        BusFactory.getDefaultBus().setProperty("org.apache.cxf.transport.http.forceVersion", "1.1");
```

Comment:
since both tests are only differing in this switch, there should be some assert that confirms the setting was actually taken into account (eg. read HTTP version from response?).

Reply:
also what about testing the default behavior to see it really resolved to HTTP/2 for WF server?

Reply:
@jbliznak I took more time than I expected to implement this. I tried with following options and none of them works:

-  add cxf logging interceptor to check the headers from the log.  None of http2 headers are logged in cxf's logging interceptor. After digging the code,  I found CXF only logs the headers it is interested like content type, content-length etc. 
-  tried to use proxy to tap the headers. Tried with different options, setup. Finally figured out it doesn't support HTTP2. The proxy wrongly change the HTTP/2 request/response to HTTP/1.1. 
After I tried this, I finally back to look at if we can check something from jdk http log . The current solution seems simple and reliable, although it is checking some internal log.